### PR TITLE
CTCTRADERS-679: Add offices of transit to ref data

### PR DESCRIPTION
**Pull request by snehaprathipat1**

### CODE REVIEW

File: app/controllers/OfficeOfTransitController.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+class OfficeOfTransitController @Inject()(officeOfTransitService: OfficeOfTransitService, cc: ControllerComponents) extends BackendController(cc) {
+
+  def officesOfTransit(): Action[AnyContent] = Action {
+
+    Ok(Json.toJson(officeOfTransitService.officesOfTransit))
+  }
+
+  def getOfficeOfTransitByText(text: String): Action[AnyContent] = Action {
+
+    Ok(Json.toJson(officeOfTransitService.getOfficesOfTransitByText(text)))
```
Comment (ashwinirudrappa) [inline app/controllers/OfficeOfTransitController.scala:36]: @snehaprathipat1 when/where we need this service, office by text

File: app/models/OfficeOfTransit.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import play.api.libs.json.Reads
+import play.api.libs.json.Writes
+import play.api.libs.json.__
+
+case class OfficeOfTransit(value: String, text: String, selected: Boolean = false)
+
+object OfficeOfTransit {
+
+  implicit def reads: Reads[OfficeOfTransit] =
+    ((__ \ "value").read[String] and
```
Comment (ashwinirudrappa) [inline app/models/OfficeOfTransit.scala:30]: @snehaprathipat1 we are planning to send auto complete json from reference data service ?

File: test/controllers/OfficeOfTransitControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import services.OfficeOfTransitService
+
+import scala.concurrent.Future
+
+class OfficeOfTransitControllerSpec extends SpecBase with MustMatchers with MockitoSugar with BeforeAndAfterEach with BeforeAndAfterAll {
+
+  val officeId = "AD000001"
+
+  private val officesOfTransit = Seq(
+    OfficeOfTransit(officeId, "SANT JULIÀ DE LÒRIA, CUSTOMS OFFICE SANT JULIÀ DE LÒRIA (AD000001)"),
```
Comment (ashwinirudrappa) [inline test/controllers/OfficeOfTransitControllerSpec.scala:44]:
```suggestion
    OfficeOfTransit(officeId, "SANT JULIÀ DE LÒRIA, CUSTOMS OFFICE SANT JULIÀ DE LÒRIA"),
```

File: test/models/OfficeOfTransitSpec.scala [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
+import org.scalatest.FreeSpec
+import org.scalatest.MustMatchers
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import play.api.libs.json._
+
+class OfficeOfTransitSpec extends FreeSpec with ScalaCheckPropertyChecks with ModelGenerators with MustMatchers {
+
+  private val officesOfTransit = OfficeOfTransit("DE009583", "Stuttgart, Stuttgart-Hauptbahnhof (DE009583)")
```
Comment (ashwinirudrappa) [inline test/models/OfficeOfTransitSpec.scala:29]: @snehaprathipat1 Please check the "Stuttgart, Stuttgart-Hauptbahnhof (DE009583)" are we expecting in this format

Comment (snehaprathipat1) [reply test/models/OfficeOfTransitSpec.scala]: No, I will update the format. Thank you!

**Review by ashwinirudrappa — APPROVED** [review]

**Pull request merged**